src/fmeval/transforms/transform_pipeline.py

```diff
@@ -0,0 +1,67 @@
+import ray.data
+from typing import List, Union
+
+from fmeval.constants import TRANSFORM_PIPELINE_MAX_SIZE
+from fmeval.exceptions import EvalAlgorithmClientError
+from fmeval.transforms.transform import Transform
+from fmeval.util import require
+
+NestedTransform = Union[Transform, "TransformPipeline"]
+
+
+class TransformPipeline:
+    def __init__(self, nested_transforms: List[NestedTransform]):
+        require(
+            isinstance(nested_transforms, List),
+            "TransformPipeline initializer accepts a list containing Transforms or TransformPipelines.",
+        )
+        seen_keys = set()
+        duplicate_keys = []
+        self.transforms: List[Transform] = []
+        for nested_transform in nested_transforms:
+            if isinstance(nested_transform, Transform):
+                for key in nested_transform.output_keys:
+                    if key in seen_keys:
+                        duplicate_keys.append(key)
+                    else:
+                        seen_keys.add(key)
+                self.transforms.append(nested_transform)
+            elif isinstance(nested_transform, TransformPipeline):
+                self.transforms += nested_transform.transforms
+            else:
+                raise EvalAlgorithmClientError(
+                    f"nested_transform has type {type(nested_transform)}, "
+                    "but either Transform or TransformPipeline is expected."
+                )
+        require(
+            len(duplicate_keys) == 0,
+            "TransformPipeline contains Transforms with the same output keys as other Transforms. "
+            f"Duplicate keys: {duplicate_keys}.",
```

Comment:
Suggestion: Instead of a returning a list of duplicate keys to the user, could we instead have a way to show which transforms share the same keys? It would help the user identify which Transforms are affected and make changes easily.

Reply:
Had the same thought just a second ago